tsl/src/continuous_aggs/options.c

```diff
@@ -62,12 +66,163 @@ update_materialized_only(ContinuousAgg *agg, bool materialized_only)
+	Relation cagg_view_rel = table_open(cagg_view_oid, AccessShareLock);
+	RuleLock *cagg_view_rules = cagg_view_rel->rd_rules;
+	Assert(cagg_view_rules && cagg_view_rules->numLocks == 1);
+	RewriteRule *rule = cagg_view_rules->rules[0];
+	if (rule->event != CMD_SELECT)
+		elog(ERROR, "rule missing for view ");
+
+	Query *cagg_view_query = copyObject(linitial(rule->actions));
+	table_close(cagg_view_rel, NoLock); // lock with be released at end of txn
+	Oid mat_relid = mat_ht->main_table_relid;
+	Query *orig_query;
+	/* the view rule has dummy old and new range table entries as the 1st and 2nd entries
+	 */
+	Assert(list_length(cagg_view_query->rtable) >= 2);
+	if (cagg_view_query->setOperations)
+	{
+		/* This corresponds to the union view.
+		 * the 3rd RTE entry has the SELECT 1 query from the union view. */
+		RangeTblEntry *orig_query_rte = lthird(cagg_view_query->rtable);
+		if (orig_query_rte->rtekind != RTE_SUBQUERY)
+			elog(ERROR, "unexpected view rule action ");
+		orig_query = orig_query_rte->subquery;
+	}
+	else
+	{
+		orig_query = cagg_view_query;
+	}
+	foreach (lc, orig_query->groupClause)
+	{
+		SortGroupClause *cagg_gc = (SortGroupClause *) lfirst(lc);
+		TargetEntry *cagg_tle = get_sortgroupclause_tle(cagg_gc, orig_query->targetList);
+		/* groupby clauses are columns from the mat hypertable */
+		Assert(IsA(cagg_tle->expr, Var));
+		Var *mat_var = castNode(Var, cagg_tle->expr);
+		char *mat_colname = get_attname(mat_relid, mat_var->varattno, false);
+		retlist = lappend(retlist, mat_colname);
+	}
+	return retlist;
+}
+
+static List *
+cagg_get_compression_params(ContinuousAgg *agg, Hypertable *mat_ht)
+{
+	List *defelems = NIL;
+	const Dimension *mat_ht_dim = hyperspace_get_open_dimension(mat_ht->space, 0);
+	const char *mat_ht_timecolname = NameStr(mat_ht_dim->fd.column_name);
+	DefElem *ordby = makeDefElemExtended("timescaledb",
+										 "compress_orderby",
+										 (Node *) makeString((char *) mat_ht_timecolname),
+										 DEFELEM_UNSPEC,
+										 -1);
+	defelems = lappend(defelems, ordby);
+	List *grp_colnames = cagg_find_groupingcols(agg, mat_ht);
+	if (grp_colnames)
+	{
+		ListCell *lc;
+		/* we have column names. they are guaranteed to be at most
+		 * NAMEDATALEN
+		 */
+		int seglen = ((NAMEDATALEN + 1) * list_length(grp_colnames)) + 1;
+		char *segmentby = (char *) palloc(seglen);
+		int segidx = 0;
+		foreach (lc, grp_colnames)
+		{
+			char *grpcol = (char *) lfirst(lc);
+			/* skip time dimension col if it appears in group-by list */
+			if (namestrcmp((Name) & (mat_ht_dim->fd.column_name), grpcol) == 0)
+				continue;
+			int collen = 1;
+			if (segidx > 0 && (seglen - segidx) > collen)
+			{
+				strlcpy(segmentby + segidx, ",", collen + 1);
+				segidx = segidx + 1;
+			}
+			collen = strlen(grpcol);
+			if (seglen - segidx > collen)
+			{
+				strlcpy(segmentby + segidx, grpcol, collen + 1);
+				segidx = segidx + collen;
+			}
+			else
+			{
+				ereport(ERROR,
```

Comment:
The code path needs a corresponding test case, if it's possible to add one. Otherwise, please add a comment that it should never happen in practice and is needed only for debugging purposes.

Reply:
Since the CI is not complaining anymore, I assume it was fixed.